src/main/scala/com/microsoft/hyperspace/telemetry/HyperspaceEventLogging.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (2020) The Hyperspace Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.microsoft.hyperspace.telemetry
+
+import scala.util.{Success, Try}
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.util.hyperspace.Utils
+
+import com.microsoft.hyperspace.telemetry.Constants._
+
+/**
+ * Hyperspace event logging interface. Extend this to enable emitting hyperspace events.
+ */
+trait HyperspaceEventLogging {
+  def logEvent(event: HyperspaceEvent): Unit = EventLogger.getLogger.logEvent(event)
+}
+
+/**
+ * Event Logger interface. Concrete implementations of this class implement event handling logic
+ * for hyperspace events.
+ */
+trait EventLogger {
+  def logEvent(event: HyperspaceEvent): Unit
+}
+
+object EventLogger extends Logging {
+  // Singleton logger instance for event logging.
+  private var logger: EventLogger = _
+
+  def getLogger: EventLogger = {
+    if (logger != null) {
+      return logger
+    }
+
+    val className: String = SparkSession.getActiveSession
```

Comment:
How about the following flow? I think if the user specified the config for the logger and if it fails to load, we should fail early instead of falling back:
```scala
    SparkSession.getActiveSession
      .map(_.conf.get(HYPERSPACE_EVENT_LOGGER_CLASS_KEY))
      .map { className =>
        Try(Utils.classForName(className).newInstance) match {
          case Success(logger: EventLogger) =>
            logInfo("")
            logger
          case Failure(exception) =>
            logError("")
            throw exception
        }
      }
      .getOrElse{
        logInfo("")
        new NoOpEventLogger
      }
  }
```